### DISCUSSION TITLE
Need an advice of how to implement a  small flow in BudiBase

### DISCUSSION THREAD

**Discussion by avifatal** [Budibase/budibase#4225] [Help]
Hi,
I need to fetch data from an api, display it in a form, let the user manipulate the data and save it in the database.
So, this is how it goes:
The user will hit a button on the screen >> 
   The button will trigger the server fetch an external api data and and display it in a form >>
      The user will be able to change the data and manipulate it  >>
          The user can save the form to a table >>

How can such think be done in BudiBase?
Thanks

**Comment by mjashanks** [comment]
HI There. This is possible, and I have created an example app for you - attached below.

So I have
1. An API call to `https://api.github.com/repos/budibase/budibase` - i.e. fetches the budibase repository details. It's named `Get Repo`
2. A table called `Repos`
3. On the `Home` screen, I have the following structure
```
> Repeater Block : Data = Get Repo
   > Form : Schema = Repo (the table), type = Create
      > Name Input : Default Value is {{ Repeater Block.Get Repo.full_name }}
      > Description Input: > Name Input : Default Value is {{ Repeater Block.Get Repo.description }}
   > Save Button : saves the row to a table
```

The key point here is that the Fields "Default Value" are bound to the values supplied by the API call.

[GH to Table-export-1643388599647.txt](https://github.com/Budibase/budibase/files/7960390/GH.to.Table-export-1643388599647.txt)
